airflow/stats.py

```diff
@@ -65,7 +65,7 @@ def gauge(cls, stat: str, value: float, rate: int = 1, delta: bool = False) -> N
         """Gauge stat"""
 
     @classmethod
-    def timing(cls, stat: str, dt: Union[float, datetime.timedelta]) -> None:
+    def timing(cls, stat: str, dt: Optional[Union[float, datetime.timedelta]]) -> None:
```

Comment:
this is not correct

Reply:
Instead you should have:

```diff
diff --git a/airflow/dag_processing/manager.py b/airflow/dag_processing/manager.py
index 408bb48aa..4483c6a58 100644
--- a/airflow/dag_processing/manager.py
+++ b/airflow/dag_processing/manager.py
@@ -70,8 +70,8 @@ class DagFileStat(NamedTuple):

     num_dags: int
     import_errors: int
-    last_finish_time: Optional[datetime]
-    last_duration: Optional[float]
+    last_finish_time: datetime
+    last_duration: float
     run_count: int
```

Reply:
The `Optional` typing was to handle this instance. Should we try something else instead?
https://github.com/apache/airflow/blob/main/airflow/dag_processing/manager.py#L1043-L1047

Reply:
In that case the following should be ok:

```diff
diff --git a/airflow/dag_processing/manager.py b/airflow/dag_processing/manager.py
index 408bb48aa..25c6ec70f 100644
--- a/airflow/dag_processing/manager.py
+++ b/airflow/dag_processing/manager.py
@@ -920,7 +920,7 @@ class DagFileProcessorManager(LoggingMixin):
         self._file_stats[processor.file_path] = stat

         file_name = os.path.splitext(os.path.basename(processor.file_path))[0].replace(os.sep, '.')
-        Stats.timing(f'dag_processing.last_duration.{file_name}', stat.last_duration)
+        Stats.timing(f'dag_processing.last_duration.{file_name}', cast(float, stat.last_duration))

     def collect_results(self) -> None:
         """Collect the result from any finished DAG processors"""
```

`Stats.timing` should never receive `None` as value - https://github.com/jsocol/pystatsd/blob/f3f304b4b2c3d5eddeb9f4977d9c82c64c37a052/statsd/client/base.py#L26-L30

Reply:
Oh OK. Right on, let me try that.

Reply:
If I move the `cast()` here we're golden:
```python
for file_path in files_paths_to_queue:
            if file_path not in self._file_stats:
                self._file_stats[file_path] = DagFileStat(
                    num_dags=0,
                    import_errors=0,
                    last_finish_time=None,
                    last_duration=cast(float, None),
                    run_count=0,
                )
```
Thanks for the tip @kaxil!